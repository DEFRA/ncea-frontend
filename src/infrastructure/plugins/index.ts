export { customHapiPino } from './logger';
export { customHapiRoutes } from './router';
export { customHapiViews } from './views';

