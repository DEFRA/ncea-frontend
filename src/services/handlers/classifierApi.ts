import { environmentConfig } from '../../config/environmentConfig';
import axios, { AxiosResponse } from 'axios';
import { classifiers, classify } from '../../interfaces/classifierSearch.interface';

const transformClassifierDetails = (classifiers: classify[]): classify[] => {
  return classifiers?.map((classifier) => ({
    ...classifier,
    text: classifier.definition,
    value: classifier.themeName || classifier.name,
  }));
};

const transformClassifierLevel3Details = (Level2Classifiers: classify[]): classifiers[] => {
  return Level2Classifiers.map((classifier2: classify) => {
    const classifiers3 = classifier2.classifiers ? transformClassifierDetails(classifier2.classifiers) : [];
    return {
      ...classifier2,
      sectionTitle: classifier2.name,
      sectionIntroduction: '',
      classifiers: classifiers3,
      selectAll: classifiers3?.map((classify) => classify.code).join(','),
      text: '',
      value: classifier2.name,
    };
  });
};

export const getClassifierThemes = async (level: string, parents: string = ''): Promise<classifiers[]> => {
  try {
    let url = `${environmentConfig.classifierApiUrl}?level=${level}`;
    if (parents) {
      url = url + `&Parents=${parents}`;
    }
    const headers = {
      headers: {
        'X-API-Key': environmentConfig.classifierApiKey,
      },
    };
    const response: AxiosResponse = await axios.get(url, headers);
    const classifierResponse: classifiers[] = [];
    response.data.forEach((classifier: classifiers) => {
      if (classifier.level === 3) {
        classifierResponse.push({
          sectionTitle: classifier.sectionTitle,
          sectionIntroduction: classifier.sectionIntroduction,
          classifiers: [],
          selectAll: '',
        });
        const lvl3: classifiers[] = transformClassifierLevel3Details(classifier.classifiers);
        classifierResponse.push(...lvl3);
      } else {
        const classifiers = transformClassifierDetails(classifier.classifiers);
        classifierResponse.push({
          sectionTitle: classifier.sectionTitle,
          sectionIntroduction: classifier.sectionIntroduction,
          classifiers,
          selectAll: classifiers.map((classify) => classify.code).join(','),
        });
      }
    });
    return classifierResponse;
  } catch (error: unknown) {
    return [];
  }
};
