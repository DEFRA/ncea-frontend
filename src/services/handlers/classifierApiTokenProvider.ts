const msal = require('@azure/msal-node');
import { environmentConfig } from '../../config/environmentConfig';
export const tokenRequest = {
    scopes: [`api://${environmentConfig.classifierApi.scopes}/.default`],
};

export const apiConfig = {
    uri: environmentConfig.classifierApi.endpoint,
};

const msalConfig = {
    auth: {
        clientId: environmentConfig.classifierApi.clientId,
        authority: `https://login.microsoftonline.com/${environmentConfig.classifierApi.authority}`,
        clientSecret: environmentConfig.classifierApi.clientSecret
    },
    system: {
        loggerOptions: {
            loggerCallback(loglevel, message, containsPii) {
                console.log(message);
            },
            piiLoggingEnabled: false,
            logLevel: 'Info',
        },
    },
};

const cca = new msal.ConfidentialClientApplication(msalConfig);

/**
 * Acquires token with client credentials.
 * @param {object} tokenRequest
 */
export const getToken = async(tokenRequest) => {
    return await cca.acquireTokenByClientCredential(tokenRequest);
}