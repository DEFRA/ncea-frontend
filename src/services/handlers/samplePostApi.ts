import { geoNetworkClient } from '../../config/geoNetworkClient';

const getSamplePosts = async () => geoNetworkClient.get('posts');

export { getSamplePosts };
