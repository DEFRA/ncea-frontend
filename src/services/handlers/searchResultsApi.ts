import { ApiResponse } from '../../Models/ApiResponse';
import { geoNetworkClient } from '../../config/geoNetworkClient';
import { BoolModel, Must, Query, Querystring, SearchRequest } from '../../Models/SearchRequest';

const url = 'https://20.77.9.229:8080/geonetwork/srv/api/search/records/_search?bucket=s101';

const getSearchRequest = function (searchTerm: string | null): SearchRequest {
  const queryString = new Querystring(searchTerm as string, 'AND');
  const must = new Must(queryString);
  const boolModel = new BoolModel([must]);
  const queryModel = new Query(boolModel);
  const searchRequestObj = new SearchRequest(queryModel, null);
  return searchRequestObj;
};

const getSearchResults = async function (query: string | null): Promise<ApiResponse> {
  try {
    const searchRequestObj = getSearchRequest(query);
    const res = await geoNetworkClient.post(url, searchRequestObj);
    return new ApiResponse(res.data, res.status, res.status == 200);
  } catch (error) {
    return new ApiResponse({ message: 'Unable to fetch the search results.' }, 400);
  }
};

export { getSearchResults };
