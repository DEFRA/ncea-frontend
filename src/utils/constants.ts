import { IFilterOptions } from '../interfaces/searchPayload.interface';
import { TabOptions } from '../interfaces/detailsTab.interface';

export const webRoutePaths = {
  home: '/',
  results: '/search',
  guidedDateSearch: '/date-search',
  geographySearch: '/coordinate-search',
  intermediate: '/intermediate',
  getMapResults: '/map-results',
  getMapFilters: '/map-filters',
  filterResourceType: '/resource-type-filter',
  filterStudyPeriod: '/study-period-filter',
  sortResults: '/sort-results',
  help: '/help',
  accessibilityStatement: '/accessibility-statement',
  termsAndConditions: '/terms-conditions',
  privacyPolicy: '/privacy-policy',
  cookiePolicy: '/cookie-policy',
};

export const elasticSearchAPIPaths = {
  searchPath: '_search',
  countPath: '_count',
};

export const formKeys = {
  dateQuestionnaire: 'date-questionnaire',
};

export const formIds = {
  quickSearchFID: 'keyword',
  dataQuestionnaireFID: 'date',
  geographyQuestionnaireFID: 'extent',
};

export const showMoreText: string = 'Show more';
export const showLessText: string = 'Show less';
export const maxWords: number = 100;
export const geoNetworkIndex: string = 'gn-records';
export const mapResultMaxCount: number = 10000;
export const requiredFieldsForMap: string[] = [
  'resourceTitleObject',
  'contactForResource',
  'resourceAbstractObject',
  'geom',
  'resourceIdentifier',
  'resourceType',
  'resourceTemporalExtentDetails',
  'ownerOrgForResourceObject',
  'originatorOrgForResourceObject',
  'custodianOrgForResourceObject',
  'distributorOrgForResourceObject',
  'pointOfContactOrgObject',
  'linkUrl',
  'OrgNceaClassifiers'
];

export const quickSearchTargetFields: string[] = [
  'metadataIdentifier',
  'resourceTitleObject.default',
  'resourceAltTitleObject.default',
  'resourceAbstractObject.default',
  'lineageObject.default',
  'tag.default',
  'cl_topic.default',
  'resourceType',
  'supplementalInformationObject.default',
];

export const guidedSearchSteps = {
  date: 'date',
  coordinate: 'extent',
};

export const queryParamKeys = {
  quickSearch: 'q',
  fromDateDay: 'fdd',
  fromDateMonth: 'fdm',
  fromDateYear: 'fdy',
  toDateDay: 'tdd',
  toDateMonth: 'tdm',
  toDateYear: 'tdy',
  coordinates: 'ext',
  north: 'nth',
  south: 'sth',
  east: 'est',
  west: 'wst',
  count: 'cnt',
  page: 'pg',
  rowsPerPage: 'rpp',
  sort: 'srt',
  journey: 'jry',
  resourceType: 'rty',
  startYear: 'sy',
  toYear: 'ty',
};

export const uniqueResourceTypesKey: string = 'unique_resource_types';
export const startYearRangeKey: string = 'start_year_range';
export const toYearRangeKey: string = 'to_year_range';
export const yearRange: string = 'year';

export const resourceTypeFilterField = 'resourceTypeFilter';
export const studyPeriodFilterField = 'studyPeriodFilter';

export const defaultFilterOptions: IFilterOptions = [
  {
    key: uniqueResourceTypesKey,
    needCount: true,
    propertyToRead: 'key',
    hasBucket: true,
    isTerm: true,
  },
  {
    key: yearRange,
    needCount: false,
    propertyToRead: 'value_as_string',
    hasBucket: false,
    isDate: true,
  },
];

export const detailsTabOptions: TabOptions = {
  general: {
    Abstract: 'content',
    'Study periods': 'studyPeriod',
    'Topic categories': 'topicCategories',
    Keywords: 'keywords',
    'Resource languages': 'language',
  },
  access: {
    'File Identifier': 'ncea_catalogue_number',
    'Resource Identifier': 'host_catalogue_number',
    'Coupled resources': 'host_catalogue_entry',
    'Resource type': 'resource_type_and_hierarchy',
    'Hierarchy level name': 'hierarchy_level',
    'Resource locators': 'resource_locators',
    'Catalouge entry': 'ncea_catalogue_number',
    'NCEA search service catalogue number': 'host_service_catalogue_number',
    'Parent ID': 'ncea_group_reference',
    'Metadata standard': 'metadata_standard',
    'Project number': 'project_number',
    'Metadata language': 'metadata_language',
  },
  quality: {
    'Date of publication': 'publicationInformation',
    'Date of creation': 'creationInformation',
    'Date of last revision': 'revisionInformation',
    'Metadata date': 'metadataDate',
    Lineage: 'lineage',
    Conformity: 'conformity',
    'Additional information': 'additionalInformation',
  },
  geography: {
    'Spatial data service': 'spatialDataService',
    'Spatial representation service': 'spatialRepresentationService',
    'Spatial referencing system': 'spatialReferencingSystem',
    'Geographic locations': 'geographicLocations',
    'Geographic boundary': 'geographicBoundaryHtml',
    map: 'geographicBoundary',
    'Vertical extent<br /><span>(Meters above sea level)</span>': 'verticalExtent',
    'Spatial resolution': 'samplingResolution',
  },
  license: {
    'Limitations on public access': 'limitation_on_public_access',
    'License constraints': 'license_constraints',
    'Distribution formats': 'available_formats',
    'Frequency of update': 'frequency_of_update',
    'Character encoding': 'character_encoding',
  },
};

export const pageTitles = {
  home: 'NCEA Search Service Home',
  date: 'NCEA Guided Search - Study Period',
  geography: 'NCEA Guided Search - Geographic Data',
  results: 'NCEA Search Results Summary',
  generalTab: 'NCEA Catalogue Detail - General',
  accessTab: 'NCEA Catalogue Detail - Access',
  naturalCapitalTab: 'NCEA Catalogue Detail - Natural Capital',
  qualityTab: 'NCEA Catalogue Detail - Quality',
  geographyTab: 'NCEA Catalogue Detail - Geography',
  governanceTab: 'NCEA Catalogue Detail - Governance',
  licenseTab: 'NCEA Catalogue Detail - License',
  help: 'Help',
  accessibility: 'Accessibility statement',
  termsAndConditions: 'Terms and conditions',
  privacyPolicy: 'Privacy policy',
  cookiePolicy: 'Cookies',
};
