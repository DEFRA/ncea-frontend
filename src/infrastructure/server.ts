'use strict';

import { environmentConfig } from '../config/environmentConfig';
import inert from '@hapi/inert';
import vision from '@hapi/vision';
import Hapi, { Server } from '@hapi/hapi';
import logger from '../infrastructure/plugins/appinsights-logger';

import { customHapiPino, customHapiRoutes, customHapiViews } from './plugins/index';

// Create the hapi server
const server: Server = Hapi.server({
  host: process.env.HOST ?? '0.0.0.0',
  port: environmentConfig.env !== 'test' ? environmentConfig.port : 4000,
  routes: {
    validate: {
      options: {
        abortEarly: false,
      },
    },
  },
});

const initializeServer = async (): Promise<Server> => {
  // Register vendors plugins
  await server.register([inert, vision]);

  // Register the custom plugins
  await server.register([customHapiViews, customHapiRoutes, customHapiPino]);

  await server.initialize();
  return server;
};

const startServer = async (): Promise<Server> => {
  logger.info('Initializing server');
  await server.start();  
  console.log(`Server running at: ${server.info.uri}`);
  return server;
};

export { initializeServer, startServer };
