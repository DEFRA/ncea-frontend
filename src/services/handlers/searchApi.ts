import { ISearchFieldsObject } from '../../models/interfaces/queryBuilder.interface';
import { buildSearchQuery } from '../../utils/queryBuilder';
import { geoNetworkAPIPaths } from '../../utils/constants';
import { geoNetworkClient } from '../../config/geoNetworkClient';

const getSearchResults = async (searchFieldsObject: ISearchFieldsObject) => {
  const payload = buildSearchQuery(searchFieldsObject);
  return geoNetworkClient.post(geoNetworkAPIPaths.elasticSearch, payload);
};

export { getSearchResults };
