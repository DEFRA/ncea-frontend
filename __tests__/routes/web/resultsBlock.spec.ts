/**
 * @jest-environment jsdom
 */

'use strict';

import { Server } from '@hapi/hapi';

import { getSearchResults } from '../../../src/services/handlers/searchApi';
import { initializeServer } from '../../../src/infrastructure/server';
import supertest from 'supertest';
import { webRoutePaths } from '../../../src/utils/constants';
import {
  searchResultsWithData,
  searchResultsWithEmptyData,
} from '../../data/searchResultsResponse';

jest.mock('../../../src/services/handlers/searchApi', () => ({
  getSearchResults: jest.fn(),
}));

jest.mock('../../../src/infrastructure/plugins/appinsights-logger', () => ({
  info: jest.fn(),
}));

let serverRequest;

const invokeRoute = async (route, payload) => {
  const response = await serverRequest.post(route).send(payload);
  const rawHTML = response.text;
  const parser = new DOMParser();
  const document = parser.parseFromString(rawHTML, 'text/html');
  return { response, document };
};

describe('Results block template', () => {
  let server: Server;

  beforeAll(async () => {
    initializeServer().then(async (s: Server) => {
      server = s;
      serverRequest = supertest(server.listener);
    });
  });

  afterAll((done) => {
    server.stop().then(() => done());
  });

  describe('Quick search journey', () => {
    describe('Search results with data', () => {
      let response;
      let document;

      beforeAll(async () => {
        (getSearchResults as jest.Mock).mockResolvedValue(
          searchResultsWithData,
        );
        const responseObject = await invokeRoute(webRoutePaths.getResults, {
          fields: { 'quick-search': { search_term: 'test' } },
        });
        response = responseObject.response;
        document = responseObject.document;
      });

      it('should match the results screen snapshot', async () => {
        expect(response.text).toMatchSnapshot();
      });

      it('should route works with status code 200', async () => {
        expect(response.statusCode).toEqual(200);
      });

      it('should render results stats properly', async () => {
        expect(
          document.querySelector('.results-stats')?.textContent?.trim(),
        ).toEqual(`${searchResultsWithData.total} results`);
      });

      it('should render the result items properly', async () => {
        const resultItemsBlock = document.querySelector(
          '.search-result__items',
        );
        const resultItems = resultItemsBlock.children;
        expect(resultItems.length).toBe(searchResultsWithData.total);
        Array.from(resultItems).forEach((resultItem: any, index) => {
          const dateValue = searchResultsWithData?.items[index]?.studyPeriod;

          expect(
            resultItem
              .querySelector('.search-result__heading')
              ?.textContent?.trim(),
          ).toEqual(searchResultsWithData?.items[index]?.title);

          expect(
            resultItem
              .querySelector('.search-result__published-label')
              ?.textContent?.trim(),
          ).toBe('Published by');
          expect(
            resultItem
              .querySelector('.search-result__published-value')
              ?.textContent?.trim(),
          ).toBe(searchResultsWithData?.items[index]?.publishedBy);

          expect(
            resultItem
              .querySelector('.search-result__study-label')
              ?.textContent?.trim(),
          ).toBe('Study period');

          expect(
            resultItem
              .querySelector('.search-result__study-value')
              ?.textContent?.trim(),
          ).toBe(dateValue);
        });
      });
    });

    describe('Search results with empty data', () => {
      let response;
      let document;

      beforeAll(async () => {
        (getSearchResults as jest.Mock).mockResolvedValue(
          searchResultsWithEmptyData,
        );
        const responseObject = await invokeRoute(webRoutePaths.getResults, {
          fields: { 'quick-search': { search_term: 'test' } },
        });
        response = responseObject.response;
        document = responseObject.document;
      });

      it('should match the results screen snapshot', async () => {
        expect(response.text).toMatchSnapshot();
      });

      it('should route works with status code 200', async () => {
        expect(response.statusCode).toEqual(200);
      });

      it('should render results stats properly', async () => {
        expect(
          document.querySelector('.results-stats')?.textContent?.trim(),
        ).toEqual(`${searchResultsWithEmptyData.total} results found`);
      });

      it('should render results empty heading properly', async () => {
        expect(
          document
            .querySelector('.search-result__heading')
            ?.textContent?.trim(),
        ).toEqual('There are no matching results');
      });

      it('should render results empty content properly', async () => {
        expect(
          document
            .querySelector('.search-result__content')
            ?.textContent?.trim(),
        ).toEqual('Search with different keywords.');
      });
    });

    describe('Search results with error', () => {
      let response;
      let document;

      beforeAll(async () => {
        (getSearchResults as jest.Mock).mockRejectedValue(
          new Error('mocked error'),
        );
        const responseObject = await invokeRoute(webRoutePaths.getResults, {
          fields: { 'quick-search': { search_term: 'test' } },
        });
        response = responseObject.response;
        document = responseObject.document;
      });

      it('should match the results screen snapshot', async () => {
        expect(response.text).toMatchSnapshot();
      });

      it('should route works with status code 200', async () => {
        expect(response.statusCode).toEqual(200);
      });

      it('should render the error message', async () => {
        expect(
          document.querySelector('.govuk-caption-m')?.textContent?.trim(),
        ).toEqual('Unable to fetch the search results. Please try again.');
      });
    });
  });

  describe('Guided search journey', () => {
    describe('Search results with empty data', () => {
      let response;
      let document;

      beforeAll(async () => {
        (getSearchResults as jest.Mock).mockResolvedValue(
          searchResultsWithEmptyData,
        );
        const responseObject = await invokeRoute(webRoutePaths.getResults, {
          fields: {
            'date-search': { 'from-date-year': '2022', 'to-date-year': '2022' },
          },
        });
        response = responseObject.response;
        document = responseObject.document;
      });

      it('should match the results screen snapshot', async () => {
        expect(response.text).toMatchSnapshot();
      });

      it('should route works with status code 200', async () => {
        expect(response.statusCode).toEqual(200);
      });

      it('should render results stats properly', async () => {
        expect(
          document.querySelector('.results-stats')?.textContent?.trim(),
        ).toEqual(`${searchResultsWithEmptyData.total} results found`);
      });

      it('should render results empty heading properly', async () => {
        expect(
          document
            .querySelector('.search-result__heading')
            ?.textContent?.trim(),
        ).toEqual('There are no matching results');
      });

      it('should render results empty content properly', async () => {
        expect(
          document
            .querySelector('.search-result__content')
            ?.textContent?.trim(),
        ).toEqual('Choose different answers to see results.');
      });

      it('should render two secondary buttons', async () => {
        expect(
          document.querySelectorAll('.govuk-button--secondary')?.length,
        ).toBe(2);
      });
    });

    describe('Search results with error', () => {
      let response;
      let document;

      beforeAll(async () => {
        (getSearchResults as jest.Mock).mockRejectedValue(
          new Error('mocked error'),
        );
        const responseObject = await invokeRoute(webRoutePaths.getResults, {
          fields: {
            'date-search': { 'from-date-year': '2022', 'to-date-year': '2022' },
          },
        });
        response = responseObject.response;
        document = responseObject.document;
      });

      it('should match the results screen snapshot', async () => {
        expect(response.text).toMatchSnapshot();
      });

      it('should route works with status code 200', async () => {
        expect(response.statusCode).toEqual(200);
      });

      it('should render the error message', async () => {
        expect(
          document.querySelector('.govuk-caption-m')?.textContent?.trim(),
        ).toEqual('Unable to fetch the search results. Please try again.');
      });
    });
  });
});
