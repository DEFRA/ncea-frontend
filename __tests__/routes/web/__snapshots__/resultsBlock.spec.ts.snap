// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results block template Guided search journey Search results with empty data should match the results screen snapshot 1`] = `
"





  <div class='govuk-section-break--visible'>
    <h2 class='govuk-heading-m results-stats govuk-!-margin-bottom-4'>
      0 results found 
    </h2>

    <div class="option-container">
      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="sort">
    Sort by
  </label>


  <select class="govuk-select option-block" id="sort" name="sort">
  
    
      
      
      <option value="best_match">Best match</option>
    
  
    
      
      
      <option value="recent_study">Most recent study</option>
    
  
  </select>
</div>


      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="page-results">
    Results per page
  </label>


  <select class="govuk-select option-items" id="page-results" name="page-results">
  
    
      
      
      <option value="20">20</option>
    
  
    
      
      
      <option value="50">50</option>
    
  
    
      
      
      <option value="100">100</option>
    
  
  </select>
</div>


    </div>
  </div>





  <div class='search-result__item'>
    <h2 class='govuk-heading-m search-result__heading search-result__heading--empty'>There are no matching results</h2>
    <p class='search-result__content search-result__content--empty'>
      
        Choose different answers to see results.
      

      
        <div class="govuk-button-group">
          
  

  
    
  

<button type="submit" class="govuk-button govuk-button--secondary" data-module="govuk-button" onclick="history.back()">
  Previous question
</button>

          
  

  
    
  

<a href="/date-search" role="button" draggable="false" class="govuk-button govuk-button--secondary" data-module="govuk-button" data-do-storage-reset="">
  Restart questionnaire search
</a>

        </div>
      
    </p>
  </div>


"
`;

exports[`Results block template Guided search journey Search results with error should match the results screen snapshot 1`] = `
"



  <p class="govuk-caption-m govuk-!-font-size-14">Unable to fetch the search results. Please try again.</p>








"
`;

exports[`Results block template Quick search journey Search results with data should match the results screen snapshot 1`] = `
"





  <div class='govuk-section-break--visible'>
    <h2 class='govuk-heading-m results-stats govuk-!-margin-bottom-4'>
      2 results
    </h2>

    <div class="option-container">
      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="sort">
    Sort by
  </label>


  <select class="govuk-select option-block" id="sort" name="sort">
  
    
      
      
      <option value="best_match">Best match</option>
    
  
    
      
      
      <option value="recent_study">Most recent study</option>
    
  
  </select>
</div>


      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="page-results">
    Results per page
  </label>


  <select class="govuk-select option-items" id="page-results" name="page-results">
  
    
      
      
      <option value="20">20</option>
    
  
    
      
      
      <option value="50">50</option>
    
  
    
      
      
      <option value="100">100</option>
    
  
  </select>
</div>


    </div>
  </div>



  <div class='search-result__items'>
    
      <div class='govuk-section-break--visible search-result__item'>
        <h2 class='govuk-heading-m search-result__heading'>Title 1</h2>
        <div class='search-result__parameter'>
          <span class='search-result__parameter-label search-result__published-label'>Published by</span>
          <span class='search-result__parameter-value search-result__published-value'>Organization 1</span>
        </div>
        
          <div class='search-result__parameter'>
            <span class='search-result__parameter-label search-result__study-label'>Study period</span>
            <span class='search-result__parameter-value search-result__study-value'>
              
                04 Jan 1960
              
            </span>
          </div>
        
        <p class='search-result__content'>
          Content 1
        </p>

        
        
          
        

        <div class='govuk-button-group search-result__button-group'>
          
  
    
  

<button type="submit" class="govuk-button" data-module="govuk-button" disabled="true">
  Open data
</button>


          
  

  
    
  

<button type="submit" class="govuk-button govuk-button--secondary" data-module="govuk-button">
  More info
</button>

        </div>
      </div>
    
      <div class='govuk-section-break--visible search-result__item'>
        <h2 class='govuk-heading-m search-result__heading'>Title 2</h2>
        <div class='search-result__parameter'>
          <span class='search-result__parameter-label search-result__published-label'>Published by</span>
          <span class='search-result__parameter-value search-result__published-value'>Organization 2</span>
        </div>
        
          <div class='search-result__parameter'>
            <span class='search-result__parameter-label search-result__study-label'>Study period</span>
            <span class='search-result__parameter-value search-result__study-value'>
              
                04 Jan 1960 to 12 Jan 2009
              
            </span>
          </div>
        
        <p class='search-result__content'>
          Content 2
        </p>

        
        
          
        

        <div class='govuk-button-group search-result__button-group'>
          
  
    
  

<button type="submit" class="govuk-button" data-module="govuk-button" disabled="true">
  Open data
</button>


          
  

  
    
  

<button type="submit" class="govuk-button govuk-button--secondary" data-module="govuk-button">
  More info
</button>

        </div>
      </div>
    
  </div>




<div id="overlay" class="overlay"></div>
<div id="modal" class="modal">
  <div class="modal-content">
    
  <div class='data-container'>
    <h1 class='govuk-heading-l data-container__heading-l govuk-!-margin-bottom-0'>You are about to leave Natural Capital Search Service</h1>
    <p class='govuk-body-l data-container__body-l govuk-!-margin-bottom-0'>You are following a link to an external catalogue hosted by:"<span id='resource_locator'></span>"</p>
    <div class='govuk-button-group data-container__button-group govuk-!-margin-bottom-0'>
      
  

  
    
  

<button type="submit" class="govuk-button govuk-button--secondary govuk-!-margin-bottom-0" data-module="govuk-button" onclick="closeDataModal()">
  Cancel
</button>

      
  

  
    
  

<a href="#" role="button" draggable="false" class="govuk-button govuk-!-margin-bottom-0" data-module="govuk-button" onclick="closeDataModal()" target="_blank" id="resource-locator-link">
  Open site
</a>

    </div>
  </div>

  </div>
</div>"
`;

exports[`Results block template Quick search journey Search results with empty data should match the results screen snapshot 1`] = `
"





  <div class='govuk-section-break--visible'>
    <h2 class='govuk-heading-m results-stats govuk-!-margin-bottom-4'>
      0 results found 
    </h2>

    <div class="option-container">
      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="sort">
    Sort by
  </label>


  <select class="govuk-select option-block" id="sort" name="sort">
  
    
      
      
      <option value="best_match">Best match</option>
    
  
    
      
      
      <option value="recent_study">Most recent study</option>
    
  
  </select>
</div>


      


<div class="govuk-form-group option-group">
  <label class="govuk-label option-label" for="page-results">
    Results per page
  </label>


  <select class="govuk-select option-items" id="page-results" name="page-results">
  
    
      
      
      <option value="20">20</option>
    
  
    
      
      
      <option value="50">50</option>
    
  
    
      
      
      <option value="100">100</option>
    
  
  </select>
</div>


    </div>
  </div>





  <div class='search-result__item'>
    <h2 class='govuk-heading-m search-result__heading search-result__heading--empty'>There are no matching results</h2>
    <p class='search-result__content search-result__content--empty'>
      
        Search with different keywords.
      

      
    </p>
  </div>


"
`;

exports[`Results block template Quick search journey Search results with error should match the results screen snapshot 1`] = `
"



  <p class="govuk-caption-m govuk-!-font-size-14">Unable to fetch the search results. Please try again.</p>








"
`;
